pageserver/src/virtual_file/owned_buffers_io/io_buf_ext.rs

```diff
@@ -29,6 +29,17 @@ where
     }
 }
 
+impl<B> Clone for FullSlice<B>
+where
+    B: IoBuf + Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            slice: self.slice.get_ref().clone().slice_full(),
+        }
+    }
+}
+
```

Comment:
While this impl here isn't wrong, I would prefer if the owned_buffered_writer defined its own trait `trait CheapCloneForRead: Clone` and require _that_ trait for the owned_buffered_writer instead of just lean on `Clone`.

```rust
trait CheapCloneForRead : Clone {
    fn cheap_clone(&self) -> Self {
        Clone::clone(self)
    }
}
```

The reason is that `Vec::clone` is expensive and the owned buffered writer is written with the assumption that the clone is cheap.
Having such a marker trait would prevent efficiency bugs from creeping in.

Also, that trait should be defined in the `owned_buffered_writer`, resulting in better reading mental locality for readers of owned_buffered_writer`.

Reply:
d4ebd5ccd31e61a1c2aca637f058fac74cf5cb40